backend/typescript/services/implementations/userService.ts

```diff
@@ -0,0 +1,282 @@
+import * as firebaseAdmin from "firebase-admin";
+
+import IUserService from "../interfaces/userService";
+import MgUser, { User } from "../../models/user.mgmodel";
+import { CreateUserDTO, UpdateUserDTO, UserDTO } from "../../types";
+import Logger from "../../utilities/logger";
+
+class UserService implements IUserService {
+  /* eslint-disable class-methods-use-this */
+  async getUserById(userId: string): Promise<UserDTO> {
+    let user: User | null;
+    let firebaseUser: firebaseAdmin.auth.UserRecord;
+
+    try {
+      user = await MgUser.findById(userId);
+
+      if (!user) {
+        throw new Error(`userId ${userId} not found.`);
+      }
+
+      firebaseUser = await firebaseAdmin.auth().getUser(user.authId);
```

Comment:
The Firebase query is just to get the email, we could store email in MongoDB as well to avoid doing this for all the gets, but decided not to prematurely optimize (also I think duplicating the data will further complicate consistency issues)

Reply:
This seems ok, as long as there are not a lot of cases where we need the email for non-auth purposes